### PR TITLE
551 nick new

### PR DESCRIPTION
**Pull request by nalanzil**
### What are the issues this pull addresses (issue numbers / links)?

This PR addresses issue #551 by creating two Josephson Junction qcomponents. This branch can be deleted upon successful merge. 

### Did you add tests to cover your changes (yes/no)?

Yes, a demo notebook tests that the qcomponents are handled appropriately. 

### Did you update the documentation accordingly (yes/no)?

I included a demo notebook as some documentation, but the documentation on the qiskit metal website still needs to be updated. 

### Did you read the CONTRIBUTING document (yes/no)?

yes

### Summary

PR for two new JJ qcpomonents and corresponding demo notebook.

### Details and comments

Two new qcomponents: a Manhattan-style JJ and a Dolan-style JJ
One new demo notebook: showcasing how to instantiate a JJ in between the pads of a transmon pocket.

### CODE REVIEW

File: qiskit_metal/qlibrary/qubits/JJ_Manhattan.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+from qiskit_metal import draw, Dict
+from qiskit_metal.qlibrary.core.base import QComponent
+import numpy as np
+
+
+class JJ_Manhattan(QComponent):
+    """
+    The base "JJ_Manhattan" inherits the "QComponent" class.
+
+    This creates a "Manhattan"-style Josephson Junction consisting
```
Comment (zlatko-minev) [inline qiskit_metal/qlibrary/qubits/JJ_Manhattan.py:26]: Awesome, maybe add a reference to a paper, such as the Dolan and Pop papers I sent over.

**Comment by zlatko-minev** [comment]
Great work. Could you add some screenshots of the resulting structures here?

File: qiskit_metal/qlibrary/qubits/JJ_Dolan.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+from qiskit_metal import draw, Dict
+from qiskit_metal.qlibrary.core.base import QComponent
+import numpy as np
+
+
+class JJ_Dolan(QComponent):
+    """
+    The base "JJ_Dolan" inherits the "QComponent" class.
+
+    This creates a "Manhattan"-style Josephson Junction consisting
```
[the remark this answers is not in the document]
Comment (nalanzil) [reply qiskit_metal/qlibrary/qubits/JJ_Dolan.py]: Done!

File: qiskit_metal/qlibrary/qubits/JJ_Manhattan.py (hunk 1 above)
Comment (nalanzil) [reply qiskit_metal/qlibrary/qubits/JJ_Manhattan.py:26]: Done!

File: tutorials/Appendix B Quick topics/JJ Demo Notebook.ipynb [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+    "Next, let's see what a Manhattan-style JJ looks like. We'll import the qcomponent and place it in our design, calling it \"JJ1\":"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from qiskit_metal.qlibrary.qubits.JJ_Manhattan import JJ_Manhattan\n",
```
Comment (JeremyDrysdale-ibm) [inline tutorials/Appendix B Quick topics/JJ Demo Notebook.ipynb:65]: Failure in this cell
image

File: tutorials/Appendix B Quick topics/JJ Demo Notebook.ipynb [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+    "Now, let's take a look at a Dolan-style JJ qcomponent design. We'll call this one \"JJ2\" and we'll place it such that the bottom left corner is at the coordinate (0.1, 0.0) so that we can look at the two qcomponents side-by-side in our design: "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from qiskit_metal.qlibrary.qubits.JJ_Dolan import JJ_Dolan\n",
```
Comment (JeremyDrysdale-ibm) [inline tutorials/Appendix B Quick topics/JJ Demo Notebook.ipynb:89]: Error in this cell
image

**Review by JeremyDrysdale-ibm — CHANGES_REQUESTED** [review]
Overall nice, but the tutorial notebook needs to be fixed.

File: tutorials/Appendix B Quick topics/JJ Demo Notebook.ipynb (hunk 4 above)
Comment (nalanzil) [reply tutorials/Appendix B Quick topics/JJ Demo Notebook.ipynb]: hmmm, I had this fixed but it looks like I'm seeing an earlier version here. Must be some version control issue. Will update now.

**Review by JeremyDrysdale-ibm — APPROVED** [review]
LGTM

**Comment by zlatko-minev** [comment]
@ThomasGM4 

In the following output 
image

wouldn't we want the top right lead to have a 90 deg bend that runs all the way up. 

The leads should each really ideally be only a few um each in width leading up to the pads - typical for the standard literature. 

These might be a few things to tweak?

**Review by zlatko-minev — DISMISSED** [review]
Approve pending @ThomasGM4  review

**Review by ThomasGM4 — APPROVED** [review]

**Review by JeremyDrysdale-ibm — APPROVED** [review]

**Pull request merged**